client/src/js/services/CashService.js

```diff
@@ -109,101 +92,59 @@ function CashService($http, util, Exchange, uuid, sessionService ) {
-    return $http.delete(target)
-      .then(util.unwrapHttpResponse);
+    // call the prototype create method with the formatted data
+    return PrototypeApiService.create.call(service, { payment : data });
   }
 
   /**
    * Searches for a cash payment by its reference.
    *
    * @method reference
-   * @param {string} reference
-   * @returns {promise} promise - a resolved or rejected promise with the
+   * @param {String} reference
+   * @returns {Promise} promise - a resolved or rejected promise with the
    * result sent from the server.
    */
   function reference(ref) {
-    var target = baseUrl + 'references/' + ref;
-
-    return $http.get(target)
-      .then(util.unwrapHttpResponse);
+    var target = service.url.concat('references/', ref);
+    return this.$http.get(target)
+      .then(this.util.unwrapHttpResponse);
   }
 
   /**
-  * This methode is responsible to create a voucher object and it back
-  **/
-  function getTransferRecord (cashAccountCurrency, amount, currency_id){
+   * This method is responsible to create a voucher object and it back
+   */
+  function getTransferRecord(cashAccountCurrency, amount, currencyId) {
     /**
      * The date field is set at the server side
      * @todo the date in timestamp type in the database
      */
     var voucher = {
-      uuid : uuid(),
-      project_id : sessionService.project.id,
-      currency_id : currency_id,
+      project_id : Session.project.id,
+      currency_id : currencyId,
       amount : amount,
       description : generateTransferDescription(),
-      user_id : sessionService.user.id,
-      items : []
-    };
-
-    var cashVoucherLine = {
-      uuid : uuid (),
-      account_id : cashAccountCurrency.account_id,
-      debit : 0,
-      credit : amount,
-      voucher_uuid : voucher.uuid
+      user_id : Session.user.id,
+
+      // two lines (debit and credit) to be recorded in the database
+      items : [{
+        account_id : cashAccountCurrency.account_id,
+        debit : 0,
+        credit : amount,
+      }, {
+        account_id : cashAccountCurrency.transfer_account_id,
+        debit : amount,
+        credit : 0,
+      }]
     };
 
-    var transferVoucherLine = {
-      uuid : uuid (),
-      account_id : cashAccountCurrency.transfer_account_id,
-      debit : amount,
-      credit : 0,
-      voucher_uuid : voucher.uuid
-    };
-
-    voucher.items.push(cashVoucherLine);
-    voucher.items.push(transferVoucherLine);
-
     return { voucher : voucher };
   }
 
   /**
-  * This methode is responsible to generate a description for the transfer operation
-  * @private
-  **/
+   * This method is responsible to generate a description for the transfer operation.
+   * @private
+   */
   function generateTransferDescription (){
-    return ['Transfer voucher', new Date().toISOString().slice(0, 10), sessionService.user.id].join('/');
+    return 'Transfer Voucher/'.concat(new Date().toISOString().slice(0, 10), '/', Session.user.id);
```

Comment:
This isn't directly relevant to the change made here. 

This date -> toISOString() -> slice operation here is pretty abstract and may well return something different on different browsers/ machines, to standardise this (auto generated descriptions) across the client and the server we could use the `moment` library. 

This also has the benefit of making the code readable.

For example 
```js
// Explicit - this will always be the same
moment().format('YYYY-MM-DD'); // 2016-05-25

// Much easier to write - changes with locale but that can be researched 
moment().format('L'); // 05/25/2016
```